uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/leave-balance-report.api.spec.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'mocks/data/leave-balance-report-data',
+  'leave-absences/shared/apis/leave-balance-report.api'
+], function (balanceReportMockData) {
+  describe('LeaveBalanceReportAPI', function () {
+    var $httpBackend, LeaveBalanceReportAPI;
+
+    beforeEach(module('leave-absences.apis'));
+    beforeEach(inject(function (_$httpBackend_, $q, _LeaveBalanceReportAPI_) {
+      $httpBackend = _$httpBackend_;
+      LeaveBalanceReportAPI = _LeaveBalanceReportAPI_;
+
+      interceptHttp();
+    }));
+
+    describe('.getAll()', function () {
+      describe('when calling .getAll()', function () {
+        var expected, result;
+
+        beforeEach(function () {
+          var values = balanceReportMockData.all().values;
+          expected = {
+            list: values,
+            total: values.length,
+            allIds: jasmine.any(String)
+          };
+
+          LeaveBalanceReportAPI.getAll().then(function (values) {
+            result = values;
+          });
+          $httpBackend.flush();
+        });
+
+        it('returns a list of records from the report', function () {
```

Comment:
it seems that by testing what the data is, you're testing the behaviour of the backend api, which is out of the scope of the test here (you are not testing the backend api, you're testing the service that communicates with it). So what you can test about the service is that it sends the correct data to the backend api, and if it transforms the returned data as expected (but i don't think you're doing any transformation in this service)